File: src/main/scala/com/rohitpothuri/rbac/workflows/GreetingWorkflowImpl.scala
package com.rohitpothuri.rbac.workflows

import io.temporal.activity.ActivityOptions
import io.temporal.spring.boot.WorkflowImpl
import io.temporal.workflow.Workflow

import java.time.Duration

@WorkflowImpl(taskQueues = Array("GreetingTaskQueue"))
class GreetingWorkflowImpl extends GreetingWorkflow {
  private[workflows] val options = ActivityOptions.newBuilder
    .setStartToCloseTimeout(Duration.ofSeconds(5))
    .build
  final private val activities =
    Workflow.newActivityStub(classOf[GreetingActivity], options)

  override def execute(name: String): String = activities.greet(name)

  /*@Override
      public Promise<String> executeAsync(String name) {
          Promise<String> hello = Async.function(activities::greet, name);
          return hello;
      }*/
}

File: src/main/scala/com/rohitpothuri/rbac/workflows/GreetingWorkflowImpl.java
package com.rohitpothuri.rbac.workflows;

import io.temporal.activity.ActivityOptions;
import io.temporal.spring.boot.WorkflowImpl;
import io.temporal.workflow.Async;
import io.temporal.workflow.Promise;
import io.temporal.workflow.Workflow;
import org.springframework.beans.factory.annotation.Autowired;

import java.time.Duration;

@WorkflowImpl(taskQueues = "GreetingTaskQueue")
public class GreetingWorkflowImpl implements GreetingWorkflow {
    ActivityOptions options = ActivityOptions.newBuilder()
            .setStartToCloseTimeout(Duration.ofSeconds(5))
            .build();

    private final GreetingActivity activities =
            Workflow.newActivityStub(GreetingActivity.class, options);
    @Override
    public String execute(String name) {
        return activities.greet(name);
    }

    /*@Override
    public Promise<String> executeAsync(String name) {
        Promise<String> hello = Async.function(activities::greet, name);
        return hello;
    }*/
}
